internal/hcsoci/hcsdoc_wcow.go

```diff
@@ -77,6 +77,55 @@ func createMountsConfig(ctx context.Context, coi *createOptionsInternal) (*mount
 	return &config, nil
 }
 
+// ConvertCPULimits handles the logic of converting and validating the containers CPU limits
+// specified in the OCI spec to what HCS expects.
+//
+// `cid` is the container's ID.
+//
+// `vmid` is the Utility VM's ID if the container we're constructing is going to belong to
+// one.
+//
+// `spec` is the OCI spec for the container.
+//
+// `maxCPUCount` is the maximum cpu count allowed for the container. This value should
+// be the number of processors on the host, or in the case of a hypervisor isolated container
+// the number of processors assigned to the guest/Utility VM.
+//
+// Returns the cpu count, cpu limit, and cpu weight in this order. Returns an error if more than one of
+// cpu count, cpu limit, or cpu weight was specified in the OCI spec as they are mutually
+// exclusive.
+func ConvertCPULimits(ctx context.Context, cid string, spec *specs.Spec, maxCPUCount int32) (int32, int32, int32, error) {
+	cpuNumSet := 0
+	cpuCount := oci.ParseAnnotationsCPUCount(ctx, spec, oci.AnnotationContainerProcessorCount, 0)
+	if cpuCount > 0 {
+		cpuNumSet++
+	}
+
+	cpuLimit := oci.ParseAnnotationsCPULimit(ctx, spec, oci.AnnotationContainerProcessorLimit, 0)
+	if cpuLimit > 0 {
+		cpuNumSet++
+	}
+
+	cpuWeight := oci.ParseAnnotationsCPUWeight(ctx, spec, oci.AnnotationContainerProcessorWeight, 0)
+	if cpuWeight > 0 {
+		cpuNumSet++
+	}
+
+	if cpuNumSet > 1 {
+		return 0, 0, 0, fmt.Errorf("invalid spec - Windows Process Container CPU Count: '%d', Limit: '%d', and Weight: '%d' are mutually exclusive", cpuCount, cpuLimit, cpuWeight)
```

Comment:
Can we remove "Process" from this strsing, as it can be used for job containers too?

Reply:
How do you catch this stuff 😆. Sure, removing